lib/partitioned/partitioned_base.rb

```diff
@@ -107,14 +112,20 @@ def self.sql_adapter
       return @sql_adapter
     end
     
-    def self.arel_table_from_key_values(partition_key_values, as = nil)
+    def self.arel_table_from_key_values(partition_key_values, as_what = nil) # rubocop:disable Metrics/MethodLength
       @arel_tables ||= {}
-      new_arel_table = @arel_tables[[partition_key_values, as]]
+      new_arel_table = @arel_tables[[partition_key_values, as_what]]
       
       unless new_arel_table
-        arel_engine_hash = {:engine => self.arel_engine, :as => as}
+        arel_engine_hash =
+          if ActiveRecord::VERSION::MAJOR < 5
+            { as: as_what, engine: arel_engine }
+          else
+            { as: as_what }
+          end
+
```

Comment:
Here's an example of how we deal with multiple Rails versions. This kind of technique was inspired by:

- https://github.blog/2018-09-28-upgrading-github-from-rails-3-2-to-5-2/